### DISCUSSION TITLE
Can't the maximum display resolution of a video exceed 8000?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#3011] [Q&A] [answered]
I use stageVideos, video. viewPort=new Rectangle (0,0,86401920); I want to play a large video, but it will prompt me
RangeError: Error # 2006: The supplied index is out of bounds
At flash. media:: StageVideo/set viewPort()
When I change 8460 to another value, such as 5000, the playback display will be normal.
Excuse me, is this a restriction? Or did I overlook something somewhere?
thanks

**Comment by RayGyoe** [comment]
Similar issues: https://github.com/airsdk/Adobe-Runtime-Support/issues/2318


https://airsdk.dev/reference/actionscript/3.0/flash/display3D/Context3D.html#ignoreResourceLimits

Attempt to set: ignoreResourceLimits？

**Reply by Jason0247** [reply]
I didn't set this parameter, I'll go and check it out first. Thank you

**Reply by Jason0247** [reply]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/91530077/473e1bbc-60f7-48a7-bedb-289872a6ee8b)
Hello, may I ask where to set this property?
This is my code snippet. I think the file should belong to the properties in Context3D, but I don't seem to have used it here.
Thank you. @RayGyoe

**Reply by RayGyoe** [reply]
Yes, the above attributes are indeed not applicable to stageVideo.

This post may better illustrate the issue：https://github.com/airsdk/Adobe-Runtime-Support/issues/196#issuecomment-842433623

**Reply by Jason0247** [reply]
I tried playing a video with a resolution of 7560 * 1920, but strangely, it would stop playing for about 2 seconds.
When I reduce the video to 7000 width again, the playback will be normal, but it is not easy to use seek (0) to control replay.
When I reduce the video to 1920 width, seek (0) can be used.

**Reply by M4rkY4** [reply, marked as answer]
Just my subjective thoughts:

StageVideo is hardware acceleration, it should be at least strictly bound to your GPU limits. Perhaps, your video resolution is bound to max texture atlas size, which might be 8192x8192 for your GPU in case of discrete GPU? Or bound to both max atlas size and RAM allowed to be wired for your GPU in case of integrated one?
Moreover, you are using a video codec, right? Most likely it has its own resolutions support (at least in hardware decoder) and your video most likely exceeds them, which might be a bottleneck in case it's decoded via hardware, though even software decoding might have their resolution limits. GPU/CPU vendor and architecture play additional role in what codecs could be decoded via hardware.
And your SSD speed might make it all worse.

While software decoding is slower, it might have less technical limits, but way more raw hardware speed requirements.

To sum, there's plenty of potential bottlenecks here. Some because of hardware acceleration used. Perhaps, video is not an option for those enormous resolutions, or at least you should have a great GPU, a nice NVME SSD and a smartly picked codec.

**Reply by Jason0247** [reply]
Yes, it may be this issue. I am using a 3060 graphics card and seeing what you said about 8192, it should be this issue.
But strangely enough, it's possible for me to play videos of 7000 * 1800, but I can't loop through them. There may be a white screen or freeze frame during each loop. Do you know the reason for this? Is my video encoding incorrect? I am outputting an H.264 MP4 video.

**Reply by M4rkY4** [reply]
On Wiki of Nvidia_NVDEC, (NVidia hardware decoding)
H.264/AVCHD seems to have limit of 4096 х 4096
H.265/HEVC has limits from 4096 х 4096 up to 8192 x 8192

If you're using H.264, perhaps what might happen is that software decoder is used, as it seems too high for GPU decoder. First thing to try is to convert video to H.265/HEVC, perhaps it will be decoded via hardware and freezes disappear.

And if you use software decoder, I might recommend to try int**ra**frame codec on NVME SSD.
Video file will use way more space on SSD, but as each frame will be encoded separately to others, it should be decoded way faster than in case of int**er**frame codecs that you use.

What also might happen is Garbage Collection, freeing memory of multiple frames may cause seek slowdown because of int**er**frame codec.

Just test with some int**ra**frame codecs like **Cineform** / **Motion JPEG** for quick result or **DNxHR** / **DNxHD** for a good image quality. At least give it a quick test if video is short, as video file size might be too big for a long video.

Alternatively, you can try to change how you reset your video, as int**er**frame codecs hate going to a specific frame unless it's a keyframe (though frame 0 should be a keyframe, but who knows what's happening there...). Perhaps, you should rewind video differently (just stop/start, not seek) or find some way to cache or pre-load as a second video.

**Reply by Jason0247** [reply]
I have used H.265, but I am unable to display the video.
For replay, I also tried to use the method of re reading the video, but this way every replay will display the background color of the stage. I originally intended to use the first image of the video as the background, then reload the video to display this image, and then hide it when the video starts playing. However, strangely enough, the video loaded by stageVide showed a color difference, which is very headache inducing. Due to the color difference, the colors of the two images do not match during replay.
#3018

**Reply by M4rkY4** [reply]
Have you tried splitting the video into two and stick them inside the app? Perhaps, two hardware-decoded videos will work better than a single software-decoded? A stitch might be visible, though.

Or check whether there's some sort of a color profile saved in video file that changes colors. Perhaps, change container format from MP4 to FLV.

In case video is simple text over flowers video, another specific option is to make a single 4096 video without text, import to Animate timeline as FLV and draw/animate vector text above it via DisplayList (Adobe Animate). Vector text sharpness might give an illusion it's a 8k video :)

**Reply by Jason0247** [reply]
My video is actually just an image without adding any configuration files. I just exported an image through AE to play a 10 second video.
What software tools can FLV use to transcode now? I have considered FLV before, but now it is difficult to find a transcoding tool.
A long time ago, FLV format videos could only reach 1920X1080, but later f4V format was released, which seems to have a much larger size.
The method of synchronously playing two videos has been considered and tested, and in some cases, there are indeed cases where the frames do not match, which is obvious.

**Reply by M4rkY4** [reply]
I've ran out of options.
Yes, FLV seems to be outdated, can't find it in Adobe Media Encoder. Previously, I'd used https://en.wikipedia.org/wiki/FormatFactory , wiki says it still supports FLV, though while installing be sure to uncheck "adware".
I can only suggest trying to show video inside a WebView, perhaps H.265 will work there.

**Reply by Jason0247** [reply]
WebView is a good idea, I'll give it a try and see if it works. Thank you for the reminder!

**Reply by itlancer** [reply]
StageWebView support right now only resolution up to 8191 pixels per side. Trying to use 8192 and more cause exception: https://github.com/airsdk/Adobe-Runtime-Support/issues/3024

**Reply by Jason0247** [reply]
If that's the case, then it seems that the webView solution is also not feasible.
I really can't think of a better way.

**Reply by Jason0247** [reply]
I created an HTML locally, but when I tried to access it using this method, the access failed. What is the problem?
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/91530077/2358f456-fa9d-44d0-a0a8-2f73eb10fed3)
The above image shows the content displayed after running
var rect:Rectangle = new Rectangle(0,0,1920,1080);
var webView:StageWebView = new StageWebView();
webView.viewPort = rect;
webView.stage = stage;
webView.loadURL("file:///1.html");

**Reply by ajwfrost** [reply]
Is this on Windows? Not sure where "file:///1.html" would go to... you'd essentially be hitting the function (if using WebView2):
https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2?view=webview2-1.0.2210.55#navigate
and it should then work the same as if you type the URL into Edge.

So e.g. for a file in the root of your C: drive, try "file://C|/1.html"

**Reply by Jason0247** [reply]
Thank you very much, this method is very useful.

**Comment by ajwfrost** [comment]
On a generic note here ... I don't know if there's any hardware/system restriction that may happen, but AIR has a soft-limit on the StageVideo and StageWebView dimensions. No documentation about why this limit exists, so we can try removing it and see what happens. One thing that you may find when pushing to larger values: the calculations for layout within the runtime tend to use twips (x20) and then there's anti-aliasing (x4) so whatever dimensions you use, we'll be having to do calculations with 80x those numbers, which may result in some overflow at some point. Although that still feels like you should have plenty of pixels to play with....

**Reply by Jason0247** [reply]
The biggest problem is actually that the video gets stuck when it gets too big. If you want to try displaying the video through stageView or HTMLLoader, you can't make it appear at the bottom layer. It always appears at the top layer, which can block other display objects.
I don't know if there's any good way?
Thank you.
#3031

**Reply by ajwfrost** [reply]
This is something we're addressing in our updated multimedia APIs... what platform are you using, is it Windows?

**Reply by Jason0247** [reply]
Yes, I am using the Windows system.
I am using AIRSDK_ 50.2.4.
Has the current version resolved the webView hierarchy issue? Will the next version solve the webView hierarchy issue? 
There are too many application requirements for using large-sized videos as dynamic base images in Windows development nowadays
If it is possible to play the video directly, it would be best. If not, looping the video through H5 is also a good way.
But currently there are two issues that need to be addressed:
The size of stageView cannot exceed 8000 pixels, which may cause errors.#3024
The hierarchy issue of stageView, if it can be kept at the bottom level, is also possible, like StageVideo.
Thank you.

**Comment by Jason0247** [comment]
Hello, may I ask which version of AIR currently has no video resolution limit?
Thank you! @ajwfrost

**Reply by ajwfrost** [reply]
If you use AIR 51.0 - and set you app descriptor to the 51.0 namespace, and publish the SWF with version 51 - then you should be free from the restrictions here.

**Reply by Jason0247** [reply]
Adobe AIR Beta release. Not for distribution.
There is currently no official version, is there?
Thank you!

**Reply by ajwfrost** [reply]
Correct, currently it's still just in the pre-release stage:
https://airsdk.harman.com/previews
I think one or two more iterations and we'll be able to release this "properly"...

thanks
